frontend/packages/multicluster-sdk/src/api/useFleetConfiguration.ts

```diff
@@ -0,0 +1,25 @@
+/* Copyright Contributors to the Open Cluster Management project */
+
+import { FleetConfiguration, UseFleetConfiguration } from '../types'
+import { useURLPoll as useFleetConfigurationPoll } from '../internal/useURLPoll'
+import { getBackendUrl } from './apiRequests'
+import { useEffect, useMemo, useState } from 'react'
+
+/**
+ * Hook that returns the fleet configuration. Checks periodically for changes in the fleet configuration.
+ *
+ * @returns {FleetConfiguration}
+ */
+
+export const useFleetConfiguration: UseFleetConfiguration = () => {
```

Comment:
Hey @jeswanke. I think combining everything in 1 hook makes it more difficult to use. Because we have a nullable object returned as part of a tuple, I have to do something like this:
```
const [fleetConfiguration, configLoaded, configError] = useFleetConfiguration();
const { isObservabilityInstalled } = fleetConfiguration ?? {};
```

This API also overlaps with `useHubClusterName` now, so there are two slightly different ways to get that information.

Currently we don't have use cases for the other bits of information returned by the backend API, but if/when we do, we can always add new hooks. I don't think the polling is strictly necessary either, as changing the hub cluster name or installing observability are rare events.